### PR TITLE
feat: finalize the billing page for unification

### LINKED ISSUE #966: Integrate API for BIlling Page
This issue is to differentiate between the existing UI-based unification work so that we can track progress on the UI side of things. Currently the UI for the billing page is all set and ready to go for the API. This issue is to make sure we circle back to hooking the UI up to the API once the billing-related API work is finished

### PR DESCRIPTION
**Pull request by Asalem1**
Closes #966 

This PR integrates the final piece of the billing page puzzle by hooking up the cancellation functionality. After lots of discussions, it was concluded that cancelled users should just be logged out after the cancellation request is made. 

Finally, this PR introduces error handling for balanceThreshold of the notification settings when a user tries to put a float as a number since the API doesn’t support that.

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=1728)

**Review by drdelambre — APPROVED** [review]
tldr you break it you buy it
